### DISCUSSION TITLE
mlagents 2.0 사용해서 학습시 --no-graphics 옵션 관련

### DISCUSSION THREAD

**Discussion by marvin-oh** [reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity#19] [General]
mlagents-learn ./drone_config/ppo.yaml --env=./RLVillage --num-envs=10 --time-scale 100 <ins>**--no-graphics**</ins>--run-id=[모델명]

이렇게 학습을 시키고 있는데, 드론 환경정보를 보았더니 카메라를 사용하던데

---
### TOTAL State 정리 (Size)
- `dec.obs[0]` : VisualObservation - 전방 카메라 (36, 64, 3)
- `dec.obs[1]` : VisualObservation - 우측 카메라 (36, 64, 3)
- `dec.obs[2]` : VisualObservation - 후방 카메라 (36, 64, 3)
- `dec.obs[3]` : VisualObservation - 좌측 카메라 (36, 64, 3)
- `dec.obs[4]` : VisualObservation - 하방 카메라 (36, 64, 3)
- `dec.obs[5]` : VectorObservation(95)
---

혹시 --no-graphics 옵션을 사용하면 안되는 건가요??

**Comment by Kyushik** [comment]
아마 이 부분은 확인을 하셔야 정확할 것으로 생각하는데 저희가 확인해본 바로 랜더링을 하지 않는 server 빌드의 경우 이미지 데이터를 얻을 수 있지만 검은 화면으로만 출력되는 결과를 확인했습니다. 아마 --no-graphics 옵션을 하는 경우에도 이미지 데이터는 나오지만 검은 화면으로만 나올 것 같습니다. 이 부분은 직접 한번 확인해주시면 좋을 것 같습니다. 만약 no-graphics 옵션을 속도 향상을 위해 사용하신거라면 이 경우 랜더링은 이루어지지 않기 때문에 속도 향상은 이루어질 것으로 생각합니다.

**Reply by marvin-oh** [reply]
그렇다면 --no--graphics 옵션을 사용하여 훈련을 진행했을 때, 
카메라 5개로부터 얻은 이미지는 학습에 아무런 영향을 끼치지 않게 되는 건가요?

**Reply by Kyushik** [reply]
아 mlagents-learn으로 학습시켜서 이미지까지 입력으로 사용하고 싶으신 경우 --no-graphics 옵션은 사용하지 않으시는 것을 추천드립니다! 이미지가 검은 화면으로만 나와서 이미지를 통한 상태 인식이 불가능하여 학습에 영향을 끼칠 수 없게 됩니다. 작은 이미지로라도 랜더링 하는 것을 추천드립니다.

**Reply by marvin-oh** [reply]
감사합니다
